agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -38,7 +38,11 @@ import (
 )
 
 const (
-	ExpectedProtectionResponseLength = 1
+	ExpectedProtectionResponseLength           = 1
+	ProtectionFailureReasonMissing             = "MISSING"
+	ProtectionFailureReasonNotManagedByService = "NOT_MANAGED_BY_SERVICE"
+	ProtectionFailureReasonTaskState           = "TASK_STOPPING_OR_STOPPED"
+	ProtectionFailureReasonBlockedDeployment   = "DEPLOYMENT_BLOCKED"
```

Comment:
Unfortunately it is not an enum. Failure only exists in ECS space and is not a thing that is general to AWS.

https://github.com/aws/aws-sdk-go/blob/e2962f353f41fe98b19be1b4b61e06fbe7c71a64/service/ecs/api.go#L13414

Reply:
Hmm that's quite unfortunate. I think `FailureReason` should be modeled  as an enum. With current design there is tight coupling between Agent implementation and Scheduler (or whatever backend components are involved) implementation. Is there anything we can do about this? This is looking really brittle to me. How would a future Backend developer know that changing a "Reason" string would break Agent?

Reply:
Where did we get these strings from?

Reply:
I hate to say this, but this failure reason is not only used by us but also customers. People using Web API will be broken if failure reason changes. These failure reasons are one way door and we should not expect any change.

Reply:
Thanks for the context here and offline. Understand that the failure object exists from before and it is not trivial to change it now. I think we should ask backend folks to at least add a comment in their code where they define these strings to caution future developers against changing it.